### DISCUSSION TITLE
Speaker Diarization Inference Time Optimization

### DISCUSSION THREAD

**Discussion by k3ybladewielder** [pyannote/pyannote-audio#1188] [Q&A]
On wav larger files with a longer duration, the inference time becomes longer and longer. In a 1-hour audio application, the inference time using the speaker diarization pipeline was 2 hours. Is it possible to optimize this inference time so that the diarization is done faster? If yes, how?

**Comment by k3ybladewielder** [comment]
How It could be used with bettertransformer, converted in ORNNX format or quantized ?

**Reply by hbredin** [reply]
Making things faster is definitely not at the top of my priority right now but I'd gladly consider a PR going into that direction.

**Reply by k3ybladewielder** [reply]
@hbredin Do you know any way to do it? Or any of those mentioned? I tested the ones I mentioned but ran into problems because the pipeline does not have training mode, it is already in inference mode and some other errors. I ask why, having positive results, I would like to contribute to the lib
